streams/src/main/java/org/apache/kafka/streams/kstream/internals/SlidingWindowedCogroupedKStreamImpl.java

```diff
@@ -113,14 +114,23 @@
                     + "]");
             }
 
-            supplier = Stores.persistentTimestampedWindowStore(
-                materialized.storeName(),
-                Duration.ofMillis(retentionPeriod),
-                Duration.ofMillis(windows.timeDifferenceMs()),
-                false
-            );
-
+            if (materialized.storeType().equals(Materialized.StoreType.IN_MEMORY)) {
+                supplier = Stores.inMemoryWindowStore(
```

Comment:
Just double checking if `inMemoryWindowStore` maps to timestamped-window store too?

Reply:
Yes, from the [java doc](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/state/Stores.java#L264-L267) in the `inMemoryWindowStore`, it said:
```
* Create an in-memory {@link 
* This store supplier can be passed into a {@link #windowStoreBuilder(WindowBytesStoreSupplier, Serde, Serde)} or
* {@link #timestampedWindowStoreBuilder(WindowBytesStoreSupplier, Serde, Serde)}.
```